pulsar-broker/src/main/java/org/apache/pulsar/client/impl/RawBatchMessageContainerImpl.java

```diff
@@ -90,6 +91,26 @@ public void setCryptoKeyReader(CryptoKeyReader cryptoKeyReader) {
         this.cryptoKeyReader = cryptoKeyReader;
     }
 
+    @Override
+    public boolean add(MessageImpl<?> msg, SendCallback callback) {
+        this.lastAddedMessageId = (MessageIdImpl) msg.getMessageId();
+        return super.add(msg, callback);
+    }
+
+    @Override
+    public boolean haveEnoughSpace(MessageImpl<?> msg) {
+        if (lastAddedMessageId == null) {
+            return super.haveEnoughSpace(msg);
+        }
+        // Keep same batch compact to same batch.
+        MessageIdImpl msgId = (MessageIdImpl) msg.getMessageId();
+        if (msgId.getLedgerId() != lastAddedMessageId.getLedgerId()
+                || msgId.getEntryId() != lastAddedMessageId.getEntryId()) {
+            return false;
+        }
+        return super.haveEnoughSpace(msg);
```

Comment:
Now, I think we should return true here and remove the below member vars. This batching should not be bounded by a specific size now.

```
this.maxNumMessagesInBatch = maxNumMessagesInBatch;
this.maxBytesInBatch = maxBytesInBatch;
```

Reply:
Yes, maybe we can set the value to `Integer.MAX_VALUE`, I will update it.

```
        this.maxNumMessagesInBatch = Integer.MAX_VALUE;
        this.maxBytesInBatch = Integer.MAX_VALUE;
```

Reply:
Updated.